deploy/lambda/auto-deployment.js

```diff
@@ -0,0 +1,118 @@
+const https = require('https');
+
+exports.handler = async function (event, context) {
+    if (!process.env.AUTO_DEPLOY_ENABLED) {
+        console.log('Auto Deploying disabled');
+        return;
+    }
+
+    try {
+        const autoDeployUsername = process.env.AUTO_DEPLOY_USERNAME;
+        const autoDeployToken = process.env.AUTO_DEPLOY_TOKEN;
+
+        if (!autoDeployUsername) {
+            console.error('No auto deploy username was provided!');
+        }
+        if (!autoDeployToken) {
+            console.error('No auto deploy token was provided!');
+        }
+
+        const versionURL = `${process.env.PROD_URL}/version`;
+        // TODO: Need to store in Microscope
+        const mainBranchURL = 'https://api.github.com/repos/atlassian/github-for-jira/branches/main';
+
+        const deployedCommitSHA = (await getRequest(versionURL)).commit;
+        const mainCommitSHA = (await getRequest(mainBranchURL)).commit.sha;
+
+        if (deployedCommitSHA !== mainCommitSHA) {
+            console.log('Changes found, starting deployment...');
+
+            // TODO: Need to update this URL with the actual one and store in Microscope
+            const pipelinesURL = 'https://api.bitbucket.org/2.0/repositories/harminder5/bitbucket-pipeline/pipelines/';
+            const body = {
+                "target": {
+                    "type": "pipeline_ref_target",
+                    "ref_type": "branch",
+                    "ref_name": "master",
+                    // TODO: Change the pattern to the actual one from GH4J
+                    "selector": {"type": "custom", "pattern": "deployment-to-prod"}
```

Comment:
Isn't this one called `deploy-to-prod` instead of `deployment-to-prod`?

Reply:
I think you're seeing the old version ,this has been updated too...

Reply:
Ops, yeah you're right, forgot to refresh